### PR TITLE
Conversion between nav_msgs/Odometry and ignition::msgs::Odometry

### PR DESCRIPTION
**Pull request by caguero**

### CODE REVIEW

File: .travis/build [hunk 1]
```diff
@@ -18,6 +18,7 @@ apt-get install -qq -y libignition-msgs4-dev \
                        python-rosdep \
                        ros-melodic-geometry-msgs \
                        ros-melodic-mav-msgs \
+                       ros-melodic-nav-msgs \
                        ros-melodic-rostest \
                        ros-melodic-sensor-msgs
```
Comment (chapulina) [inline .travis/build]: Do we need to list the ROS deps here even though we're using `rosdep install` below?

File: ros1_ign_bridge/test/test_utils.h [hunk 2]
```diff
@@ -1201,27 +1248,31 @@ namespace testing
...
 
   /// \brief Compare a message with the populated for testing.
   /// \param[in] _msg The message to compare.
-  void compareTestMsg(const ignition::msgs::Twist &_msg)
+  void compareTestMsg(const ignition::msgs::Odometry &_msg)
   {
-    compareTestMsg(_msg.linear());
-    compareTestMsg(_msg.angular());
+    ignition::msgs::Odometry expected_msg;
+    createTestMsg(expected_msg);
```
Comment (chapulina) [inline ros1_ign_bridge/test/test_utils.h]: `expected_msg` is not used

Comment (caguero) [reply ros1_ign_bridge/test/test_utils.h]: > `expected_msg` is not used

Added some cleans in 82e7c0a

File: .travis/build (hunk 1 above)
Comment (caguero) [reply .travis/build]: > Do we need to list the ROS deps here even though we're using `rosdep install` below?

I don't think so.  ced631f

**Comment by chapulina** [comment]
@caguero , I took the liberty of committing a demo directly to this branch: 185533d579d05c62b07020580005801f51aba120

I think all we're missing now is the `child_link_id` from `ign-gazebo`.

**Comment by caguero** [comment]
> @caguero , I took the liberty of committing a demo directly to this branch: [185533d](https://github.com/osrf/ros1_ign/commit/185533d579d05c62b07020580005801f51aba120)
> 
> I think all we're missing now is the `child_link_id` from `ign-gazebo`.

How dare you! ;) Thanks!

**Comment by nkoenig** [comment]
+1

**Pull request merged**